tests/test_healtcheck.py

```diff
@@ -0,0 +1,169 @@
+import json
+import logging
+import random
+from time import sleep
+
+import plumbum.commands.processes
+import pytest
+from plumbum import local
+from plumbum.cmd import docker, docker_compose
+
+PROXY_TARGET_PAIRS = [
+    ("proxy_preresolve", "target"),
+    ("proxy_smtp", "target_smtp"),
+    ("proxy_without_preresolve", "target"),
+]
+
+logger = logging.getLogger()
+
+
+def _healthcheck(*args, **kwargs):
+    args = ("-f", "tests/healthcheck.yaml") + args
+    return docker_compose(*args, **kwargs)
+
+
+def _get_container_id(service_name):
+    return _healthcheck("ps", "-q", service_name).strip()
```

Comment:
I don't know if you knew this, but plumbum supports these kind of constructs out of the box (it would be equivalent (except the `strip()` part)):

```suggestion
_healthcheck = docker_compose["-f", "tests/healthcheck.yaml"]
_get_container_id = _healthcheck["ps", "-q", service_name]
```

Reply:
```
_get_container_id = _healthcheck["ps", "-q", service_name]
```
does not work, because this syntax doesn't support args:
```
 NameError: name 'service_name' is not defined
```

Reply:
Ah yes, sorry, my butter fingers. It should be `_get_container_id = _healthcheck["ps", "-q"]` instead, and then later called as `_get_container_id(service_name)`

Reply:
i changed it to
```
_healthcheck["ps", "-q"]
```